pkg/parser/receiver_generic.go

```diff
@@ -0,0 +1,40 @@
+package parser
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+const parserNameGeneric = "__generic"
+
+var _ ReceiverParser = &GenericReceiver{}
+
+// GenericReceiver is a special parser for generic receivers. It doesn't self-register and should be created/used directly
+type GenericReceiver struct {
+	name   string
+	config map[interface{}]interface{}
+}
+
+// NewGenericReceiverParser builds a new parser for generic receivers
+func NewGenericReceiverParser(name string, config map[interface{}]interface{}) ReceiverParser {
```

Comment:
Should this also take a default port? So if the endpoint is not defined it could still return a port if the default is specified.

Reply:
You mean, so that similar receivers could reuse this generic parser? Makes sense, let's see what the next parsers look like.

Reply:
Done in the commit that introduced the Zipkin parsers.